docs: add general guidance on contributor expectations (#83)
cilium/community: CONTRIBUTOR-LADDER.md

**Note:** Devstats does not take into account non-code contributions. If a non-code contributing member is accidentally removed this way, they may open an issue to quickly be re-instated.

> ## General guidance on making contributions and expectations

Reviewer: Thanks for putting this together Liz. I think that this guidance could be useful for community members and I agree . Having read Nadia Eghbal's book _Working in Public: The Making and Maintenance of Open Source Software_, I think this reflects the dynamics in play in an open community.

I do wonder a little about a couple of things:
- Whether placing the guidance in this location will reach the target contributor audience better than an alternative location
- Whether this guidance provides sufficient encouragement to reassure potential contributors about how important their contributions are; for instance, regularly contributing to review can help to share the load and improve PR response time. Ideally if this statement is saying "we try to facilitate community contributions, but we don't have all the time in the world" and contributors will feel empowered to respond with "how can I help?" and not "oh okay I won't contribute then".

Author: > Whether placing the guidance in this location will reach the target contributor audience better than an alternative location

The other option I can think of would be the [Contributor Guide](https://docs.cilium.io/en/stable/contributing/development/). I'm imagining that guide will eventually [move into the community repo](https://github.com/cilium/community/issues/78) along with the governance-related sections of the docs, so that it all eventually lands in one place. I will confess that I proposed adding this text here in the meantime, because it's easier than changing the docs, but maybe that is too much laziness on my part :-D 

> contributors will feel empowered to respond with "how can I help?" and not "oh okay I won't contribute then"

Very good point. I made some text edits to attempt to address this, wdyt?

> We welcome people to raise PRs, issues and CFPs but please bear in mind that it takes time and effort to review them properly, and as an open source project we don’t make any guarantees about how quickly they will get looked at. Committers and reviewers may have other pressures on their time that mean they don’t currently have the bandwidth to look at your proposal. 

Author: ```suggestion
We welcome people to raise PRs, issues and CFPs in the Cilium project. If you're new to the project we'd strongly encourage you to start off with small, but meaningful, changes first, as these are easier and quicker for committers / reviewers to assess and give feedback on. You'll get the opportunity to learn about the [process and expectations](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/) of making changes within the project. 

As an open source project we don’t make any guarantees about how quickly proposed changes will get looked at. Committers and reviewers may have other pressures on their time, that mean they don’t currently have the bandwidth to look at your proposal. 
```

> Naturally, the more an individual has already contributed to the project, the higher the level of trust and confidence that the person will have established in the community. Starting with smaller proposals, fixes and maintenance work can help build up that trust and confidence. 

Author: ```suggestion
Naturally, the more an individual has already contributed to the project, the higher the level of trust and confidence that the person will have established in the community. Starting with smaller proposals, fixes and maintenance work, or making constructive and meaningful review comments on other people's suggestions, can help build up that trust and confidence. 
```

**Note:** Devstats does not take into account non-code contributions. If a non-code contributing member is accidentally removed this way, they may open an issue to quickly be re-instated.

## General guidance on making contributions and expectations

We welcome people to raise PRs, issues and CFPs in the Cilium project. If you're new to the project we'd strongly encourage you to start off with small, but meaningful, changes first, as these are easier and quicker for committers / reviewers to assess and give feedback on. You'll get the opportunity to learn about the [process and expectations](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/) of making changes within the project. 

> As an open source project we don’t make any guarantees about how quickly proposed changes will get looked at. Committers and reviewers may have other pressures on their time, that mean they don’t currently have the bandwidth to look at your proposal. 

Reviewer: What would you suggest community members do if committers and reviewers are persistently lacking bandwidth?

I would think that there is some path here around building alignment with similarly-minded community members to collaborate and figure out the details. Or for something bigger, following up in dedicated SIG channels on Slack, raising discussion on the #development channel, and/or bringing up the topic in community meetings.

> As an open source project we don’t make any guarantees about how quickly proposed changes will get looked at. Committers and reviewers may have other pressures on their time, that mean they don’t currently have the bandwidth to look at your proposal. 

Author: ```suggestion
As an open source project we don’t make any guarantees about how quickly proposed changes will get looked at. Committers and reviewers may have other pressures on their time, that mean they don’t currently have the bandwidth to look at your proposal. We do encourage you to discuss your ideas in Slack (in the #development channel or in the relevant SIG channel) so that similarly-minded community members can collaborate with you on the proposal.  
```

Reviewer: Looks good :+1:
